air/src/execution_step/boxed_value/canon_stream.rs

```diff
@@ -47,6 +47,7 @@ impl CanonStream {
     }
 
     pub(crate) fn from_stream(stream: &Stream, peer_pk: String, position: TracePos) -> Self {
+        // it's always safe to iter over all generations of a stream
```

Comment:
`safe` has very specific meaning in Rust...
BTW, perhaps, one should introduce `iter_all` method that doesn't return an iterator.
Or, one can call `.iter().flatten()` on the option.

Reply:
Fixed comment, re `iter_all- It'd add it in a separate PR.